lib/src/type_utils.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+extension DartTypeExtension on DartType {
+  /// Returns the 'element' for this type, which is sometimes `null`.
+  TypeDefiningElement? get element {
+    final self = this;
+    if (self is InterfaceType) {
+      return self.element2;
+    } else if (self is DynamicType) {
+      // This may be made available in analyzer.
+      // ignore: deprecated_member_use
+      return self.element as TypeDefiningElement;
```

Comment:
You cannot do this.
I will remove `DartType.element`.
Deprecation is to let you know that it will be gone, not to be ignored.

Reply:
Should I return `(self as DynamicTypeImpl).element`?

Reply:
This will still fail at runtime.
Do you need this element?

Reply:
Maybe not... it was maybe only needed in one spot, and I refactored. Uploaded a new change.